src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -25,6 +26,96 @@ public static class Renamer
                 newName, optionSet, cancellationToken);
         }
 
+        /// <summary>
+        /// Similar to calling <see cref="Document.WithName(string)" /> with additional changes to the solution. 
+        /// Each change is added as a <see cref="RenameDocumentAction"/> in the returned <see cref="RenameDocumentActionSet.ApplicableActions" />.
+        /// 
+        /// Each action may individually encounter errors that prevent it from behaving correctly. Those are reported in <see cref="RenameDocumentAction.GetErrors(System.Globalization.CultureInfo?)"/>.
+        /// 
+        /// Current supported actions that may be returned: 
+        /// * <see cref="RenameSymbolDocumentAction"/> that will rename the type to match the document name
+        /// </summary>
+        public static async Task<RenameDocumentActionSet> RenameDocumentNameAsync(
+            Document document,
+            string newDocumentName,
+            OptionSet optionSet,
+            CancellationToken cancellationToken = default)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            if (newDocumentName == null)
+            {
+                throw new ArgumentNullException(nameof(newDocumentName));
+            }
+
+            var actions = new ArrayBuilder<RenameDocumentAction>();
```

Comment:
```suggestion
            var actions = new ArrayBuilder<RenameDocumentAction>();
```

Don't new up an arraybuilder.  use ArrayBuilder.Getinstance